File: JavaScript/487-Max Consecutive Ones II.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var findMaxConsecutiveOnes = function(nums) {
    let left = 0, numZero = 0, res = 0;
    for (let i=0; i<nums.length; i++){
        if (nums[i] === 0) numZero++;
        if (numZero === 2){
            res = Math.max(res, i-1-left+1);
            while (numZero === 2){
                numZero -= 1-nums[left];
                left++;
            }
        }
    }
    res = Math.max(res, nums.length-1-left+1);
    return res;
};

File: C++/487-Max Consecutive Ones II.cpp
class Solution {
public:
    int findMaxConsecutiveOnes(vector<int>& nums) {
        int left = 0;
        int res = 0;
        int currZero = 0;
        for (int i=0; i<nums.size(); i++){
            if (nums[i] == 0) currZero++;
            if (currZero > 1){
                if (nums[left] == 0){
                    currZero--;
                }
                left++;
            }
            res = max(res, i-left+1);
        }
        return res;
    }
};